### ISSUE TITLE
[Bug]: Upgrading timescale to 2.10.0 on PostgreSQL12 distro/version: ubuntu/focal

### ISSUE THREAD

**Issue by pushpeepkmonroe** [timescale/timescaledb#5624]
### What type of bug is this?

Configuration, Incorrect result, Unexpected error, Other

### What subsystems and features are affected?

Other

### What happened?

\d _timescaledb_internal._materialized_hypertable_185
      Table "_timescaledb_internal._materialized_hypertable_185"
  Column   |           Type           | Collation | Nullable | Default
-----------+--------------------------+-----------+----------+---------
 time      | timestamp with time zone |           | not null |
 account   | text                     |           |          |
 campaign  | text                     |           |          |
 creative  | text                     |           |          |
 offer     | text                     |           |          |
 segment   | text                     |           |          |
 website   | text                     |           |          |
 doubleTap | boolean                  |           |          |
 agg_9_9   | bytea                    |           |          |
 chunk_id  | integer                  |           |          |
Indexes:
    "_materialized_hypertable_185_account_bucket_idx" btree (account, "time" DESC)
    "_materialized_hypertable_185_bucket_idx" btree ("time" DESC)
    "_materialized_hypertable_185_campaign_bucket_idx" btree (campaign, "time" DESC)
    "_materialized_hypertable_185_doubleTap_bucket_idx" btree ("doubleTap", "time" DESC)
    "_materialized_hypertable_185_offer_bucket_idx" btree (offer, "time" DESC)
    "_materialized_hypertable_185_segment_bucket_idx" btree (segment, "time" DESC)
    "_materialized_hypertable_185_website_bucket_idx" btree (website, "time" DESC)
Triggers:
    ts_insert_blocker BEFORE INSERT ON _timescaledb_internal._materialized_hypertable_185 FOR EACH ROW EXECUTE FUNCTION _timescaledb_internal.insert_blocker()
Number of child tables: 152 (Use \d+ to list them.)
 ALTER EXTENSION timescaledb UPDATE to '2.10.0';
ERROR:  column "bucket" does not exist
CONTEXT:  SQL statement "ALTER TABLE _timescaledb_internal._materialized_hypertable_185 RENAME COLUMN bucket TO "time""
PL/pgSQL function _timescaledb_internal.alter_table_column(regclass,regclass,name,name) line 3 at EXECUTE
SQL statement "CALL _timescaledb_internal.alter_table_column(user_view, mat_table, old_name, new_name)"
PL/pgSQL function inline_code_block line 18 at CALL

***************************************************************************************************************************************

Last week I was able to upgrade to the new timescale extension 2.10.0 on PostgreSQL12 with no issue using the below repos:

https://packagecloud.io/timescale/timescaledb/packages/ubuntu/focal/timescaledb-2-2.10.0-postgresql-12_2.10.0~ubuntu20.04_arm64.deb?distro_version_id=210

https://packagecloud.io/timescale/timescaledb/packages/ubuntu/focal/timescaledb-2-postgresql-12_2.10.0~ubuntu20.04_arm64.deb?distro_version_id=210

***************************************************************************************************************************************

I was also receiving the error: wrong file extension for this file: timescaledb-2-2.10.0-postgresql-12_2.10.0~ubuntu20.04_arm64.deb and timescaledb-2-postgresql-12_2.10.0~ubuntu20.04_arm64.deb

### TimescaleDB version affected

2.4.0

### PostgreSQL version used

12.14

### What operating system did you use?

Ubuntu 18.04.5 LTS

### What installation method did you use?

Other

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
sudo su - postgres
psql -X
\c <database>
ALTER EXTENSION timescaledb UPDATE to '2.10.0';
ERROR:  column "bucket" does not exist
CONTEXT:  SQL statement "ALTER TABLE _timescaledb_internal._materialized_hypertable_185 RENAME COLUMN bucket TO "time""
PL/pgSQL function _timescaledb_internal.alter_table_column(regclass,regclass,name,name) line 3 at EXECUTE
SQL statement "CALL _timescaledb_internal.alter_table_column(user_view, mat_table, old_name, new_name)"
PL/pgSQL function inline_code_block line 18 at CALL
```

### How can we reproduce the bug?

```bash
DROP VIEW IF EXISTS public.hypertable_stats cascade;
DROP VIEW IF EXISTS public.hypertable_stats_simple cascade;
DROP VIEW IF EXISTS public.hypertable_indexes cascade;
DROP VIEW IF EXISTS public.rename_tables cascade;
DROP VIEW IF EXISTS public.user_view cascade;
DROP VIEW IF EXISTS public.direct_view cascade;
DROP VIEW IF EXISTS public.partial_view cascade;
DROP VIEW IF EXISTS public.hypertable_sizes_and_retention cascade;

dpkg -l | grep -E 'postgresql|timescale'
ii  pgdg-keyring                       2018.2                              all          keyring for apt.postgresql.org
ii  postgresql-12                      12.6-1.pgdg18.04+1                  amd64        The World's Most Advanced Open Source Relational Database
ii  postgresql-client-12               12.6-1.pgdg18.04+1                  amd64        front-end programs for PostgreSQL 12
ii  postgresql-client-common           225.pgdg18.04+1                     all          manager for multiple PostgreSQL client versions
ii  postgresql-common                  225.pgdg18.04+1                     all          PostgreSQL database-cluster manager
ii  timescaledb-2-2.4.0-postgresql-12  2.4.0~ubuntu18.04                   amd64        An open-source time-series database based on PostgreSQL, as an extension.
ii  timescaledb-2-loader-postgresql-12 2.4.0~ubuntu18.04                   amd64        The loader for TimescaleDB to load individual versions.
ii  timescaledb-2-postgresql-12        2.3.1~ubuntu18.04                   amd64        An open-source time-series database based on PostgreSQL, as an extension.
ii  timescaledb-tools                  0.11.0~ubuntu18.04                  amd64        A suite of tools that can be used with TimescaleDB.

dpkg -L timescaledb-2-postgresql-12
/.
/usr
/usr/lib
/usr/lib/postgresql
/usr/lib/postgresql/12
/usr/lib/postgresql/12/lib
/usr/lib/postgresql/12/lib/timescaledb-1.7.0.so
/usr/lib/postgresql/12/lib/timescaledb-1.7.1.so
/usr/lib/postgresql/12/lib/timescaledb-1.7.2.so
/usr/lib/postgresql/12/lib/timescaledb-1.7.3.so
/usr/lib/postgresql/12/lib/timescaledb-1.7.4.so
/usr/lib/postgresql/12/lib/timescaledb-1.7.5.so
/usr/lib/postgresql/12/lib/timescaledb-2.0.0-rc1.so
/usr/lib/postgresql/12/lib/timescaledb-2.0.0-rc2.so
/usr/lib/postgresql/12/lib/timescaledb-2.0.0-rc3.so
/usr/lib/postgresql/12/lib/timescaledb-2.0.0-rc4.so
/usr/lib/postgresql/12/lib/timescaledb-2.0.0.so
/usr/lib/postgresql/12/lib/timescaledb-2.0.1.so
/usr/lib/postgresql/12/lib/timescaledb-2.0.2.so
/usr/lib/postgresql/12/lib/timescaledb-2.1.0.so
/usr/lib/postgresql/12/lib/timescaledb-2.1.1.so
/usr/lib/postgresql/12/lib/timescaledb-2.2.0.so
/usr/lib/postgresql/12/lib/timescaledb-2.2.1.so
/usr/lib/postgresql/12/lib/timescaledb-2.3.0.so
/usr/lib/postgresql/12/lib/timescaledb-2.3.1.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-1.7.0.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-1.7.1.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-1.7.2.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-1.7.3.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-1.7.4.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-1.7.5.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.0.0-rc1.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.0.0-rc2.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.0.0-rc3.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.0.0-rc4.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.0.0.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.0.1.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.0.2.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.1.0.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.1.1.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.2.0.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.2.1.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.3.0.so
/usr/lib/postgresql/12/lib/timescaledb-tsl-2.3.1.so
/usr/share
/usr/share/doc
/usr/share/doc/timescaledb-2-postgresql-12
/usr/share/doc/timescaledb-2-postgresql-12/changelog.gz
/usr/share/doc/timescaledb-2-postgresql-12/copyright
/usr/share/postgresql
/usr/share/postgresql/12
/usr/share/postgresql/12/extension
/usr/share/postgresql/12/extension/timescaledb--1.1.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.1.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.2.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.2.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.2.2--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.3.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.3.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.3.2--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.4.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.4.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.4.2--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.5.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.5.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.6.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.6.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.0.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.2--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.2.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.3--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.3.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.4--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.4.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.5--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--1.7.5.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.0-rc1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.0-rc2--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.0-rc3--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.0-rc4--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.0.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.2--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.0.2.sql
/usr/share/postgresql/12/extension/timescaledb--2.1.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.1.0.sql
/usr/share/postgresql/12/extension/timescaledb--2.1.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.1.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.2.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.2.0.sql
/usr/share/postgresql/12/extension/timescaledb--2.2.1--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.2.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.3.0--2.3.1.sql
/usr/share/postgresql/12/extension/timescaledb--2.3.0.sql
/usr/share/postgresql/12/extension/timescaledb--2.3.1.sql

 \dx
                                         List of installed extensions
        Name        | Version |   Schema   |                            Description
--------------------+---------+------------+-------------------------------------------------------------------
 pg_stat_statements | 1.7     | public     | track execution statistics of all SQL statements executed
 plpgsql            | 1.0     | pg_catalog | PL/pgSQL procedural language
 timescaledb        | 2.4.0   | public     | Enables scalable inserts and complex queries for time-series data
(3 rows)


sudo su - 
cd /etc/apt/sources.list.d
ls -larth
total 40K
-rw-r--r-- 1 root root  154 Aug  5  2021 timescale-ubuntu-timescaledb-ppa-bionic.list.save
-rw-r--r-- 1 root root   62 Aug  5  2021 pgdg.list.save
-rw-r--r-- 1 root root   54 Aug  5  2021 influxdata.list.save
-rw-r--r-- 1 root root  156 Aug  5  2021 timescale-ubuntu-timescaledb-ppa2-bionic.list
-rw-r--r-- 1 root root  154 Aug  5  2021 timescale-ubuntu-timescaledb-ppa-bionic.list
-rw-r--r-- 1 root root   62 Aug  5  2021 pgdg.list
-rw-r--r-- 1 root root   54 Aug  5  2021 influxdata.list
drwxr-xr-x 7 root root 4.0K Dec 20 20:43 ..
-rw-r--r-- 1 root root   75 Dec 20 20:43 pganalyze_collector.list
drwxr-xr-x 2 root root 4.0K Dec 20 20:43 .

modify files: timescaledb_timescale.list, timescaledb.list, timescale-ubuntu-timescaledb-ppa2-bionic.list, and timescale-ubuntu-timescaledb-ppa-bionic.list

cat timescale_timescaledb.list
# this file was generated by packagecloud.io for
# the repository at https://packagecloud.io/timescale/timescaledb

#deb [signed-by=/etc/apt/keyrings/timescale_timescaledb-archive-keyring.gpg] https://packagecloud.io/timescale/timescaledb/ubuntu/ bionic main
#deb-src [signed-by=/etc/apt/keyrings/timescale_timescaledb-archive-keyring.gpg] https://packagecloud.io/timescale/timescaledb/ubuntu/ bionic main

deb [signed-by=/etc/apt/keyrings/timescale_timescaledb-archive-keyring.gpg] https://packagecloud.io/timescale/timescaledb/ubuntu/ focal main
deb-src [signed-by=/etc/apt/keyrings/timescale_timescaledb-archive-keyring.gpg] https://packagecloud.io/timescale/timescaledb/ubuntu/ focal main

cat timescaledb.list
#deb https://packagecloud.io/timescale/timescaledb/debian/ bionic main
deb https://packagecloud.io/timescale/timescaledb/debian/ focal main

cat timescale-ubuntu-timescaledb-ppa2-bionic.list
#deb http://ppa.launchpad.net/timescale/timescaledb-ppa2/ubuntu bionic main
#deb-src http://ppa.launchpad.net/timescale/timescaledb-ppa2/ubuntu bionic main

deb http://ppa.launchpad.net/timescale/timescaledb-ppa2/ubuntu focal main
deb-src http://ppa.launchpad.net/timescale/timescaledb-ppa2/ubuntu focal main

cat timescale-ubuntu-timescaledb-ppa-bionic.list
#deb http://ppa.launchpad.net/timescale/timescaledb-ppa/ubuntu bionic main
#deb-src http://ppa.launchpad.net/timescale/timescaledb-ppa/ubuntu bionic main

deb http://ppa.launchpad.net/timescale/timescaledb-ppa/ubuntu focal main
deb-src http://ppa.launchpad.net/timescale/timescaledb-ppa/ubuntu focal main

curl -s https://packagecloud.io/install/repositories/timescale/timescaledb/script.deb.sh | os=ubuntu dist=focal sudo bash
wget --content-disposition "https://packagecloud.io/timescale/timescaledb/packages/ubuntu/focal/timescaledb-2-2.10.0-postgresql-12_2.10.0~ubuntu20.04_arm64.deb/download.deb?distro_version_id=210"
wget --content-disposition "https://packagecloud.io/timescale/timescaledb/packages/ubuntu/focal/timescaledb-2-postgresql-12_2.10.0~ubuntu20.04_arm64.deb/download.deb?distro_version_id=210"
wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo tee /etc/apt/trusted.gpg.d/pgdg.asc &>/dev/null
wget --quiet -O - https://packagecloud.io/timescale/timescaledb/gpgkey | sudo apt-key add -
wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys D8FF8E1F7DF8B07E
sudo apt-get install timescaledb-2-postgresql-12=2.10.0~ubuntu20.04
sudo apt-get install timescaledb-2-2.10.0-postgresql-12=2.10.0~ubuntu20.04

sudo apt-get update
sudo apt update
sudo apt install timescaledb-2-postgresql-12

sudo apt-cache search postgresql | grep 2.10.0
timescaledb-2-2.10.0-postgresql-12 - An open-source time-series database based on PostgreSQL, as an extension.
timescaledb-2-2.10.0-postgresql-13 - An open-source time-series database based on PostgreSQL, as an extension.
timescaledb-2-2.10.0-postgresql-14 - An open-source time-series database based on PostgreSQL, as an extension.
timescaledb-2-2.10.0-postgresql-15 - An open-source time-series database based on PostgreSQL, as an extension.
timescaledb-2-oss-2.10.0-postgresql-12 - An open-source time-series database based on PostgreSQL, as an extension.
timescaledb-2-oss-2.10.0-postgresql-13 - An open-source time-series database based on PostgreSQL, as an extension.
timescaledb-2-oss-2.10.0-postgresql-14 - An open-source time-series database based on PostgreSQL, as an extension.
timescaledb-2-oss-2.10.0-postgresql-15 - An open-source time-series database based on PostgreSQL, as an extension.

sudo apt-get install postgresql-12

sudo su - postgres
psql
\c pushnami_stats
\dx

ALTER TABLE _timescaledb_internal._partial_view_288 RENAME agg_9_9 to pstag_android;
ALTER TABLE _timescaledb_internal._direct_view_288 RENAME COLUMN count TO pstag_android;
SELECT timescaledb_pre_restore();
ALTER TABLE _timescaledb_internal._materialized_hypertable_288 RENAME COLUMN agg_9_9 to pstag_android;
ALTER TABLE _timescaledb_internal._materialized_hypertable_185 RENAME COLUMN bucket TO time;
SELECT timescaledb_post_restore();

exit

psql -X
\c pushnami_stats
ALTER EXTENSION timescaledb UPDATE to '2.10.0';
\dx

exit
```

**Comment by fabriziomello** [comment]
@pushpeepkmonroe the problem is you can't rename columns of internal caggs objects. Despite it was allowed to you rename columns for the materialization hypertable in 2.4.0 we ended up blocking this and instead implementing the correct behavior in 2.5.0 (See the commit https://github.com/timescale/timescaledb/commit/592e0bd46e). I strongly recommend you:
1. Keep the original names for internal objects
Update your extension
Migrate your old CAgg format for the new one available since 2.7.0 (more info how to do it here https://docs.timescale.com/use-timescale/latest/continuous-aggregates/migrate/)
Rename your columns as you want but using ALTER MATERIALIZED VIEW ... RENAME COLUMN... instead of trying to do it in the materialization hypertable because it is blocked now

**Comment by pushpeepkmonroe** [comment]
Thank you so much.
